docs/ref/models/database-functions.txt

```diff
@@ -819,3 +819,145 @@ Usage example::
      'minute': 'minute': datetime.datetime(2014, 6, 15, 14, 30, tzinfo=<UTC>),
      'second': datetime.datetime(2014, 6, 15, 14, 30, 50, tzinfo=<UTC>)
     }
+
+.. _window functions:
+
+Window functions
+~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2.0
+
+In addition to aggregate functions, e.g., ``Sum`` and ``Count``, there is a
+number of functions to use in a window expression
+:class:`~django.db.models.window.Window` for computing the rank of elements,
+computing the :class:`Ntile` of some rows. Some of these functions do not
+necessarily take any input arguments. Other expressions such as ``WITHIN
+GROUP`` may specify an argument, for instance in :class:`Rank`.
+
+.. note::
+
+    PostgreSQL does not support ``RESPECT NULLS`` or ``IGNORE NULLS`` option
+    for ``Lead``, ``Lag``, ``FirstValue``, ``LastValue``, and ``NthValue``.
+    The default is ``RESPECT NULLS``. For ``NthValue``, it is not possible to
+    use ``FROM FIRST`` or ``FROM LAST``, although a workaround is to reverse
+    the ordering of the window.
+
+``CumeDist``
+------------
+
+.. class:: CumeDist(*expressions)
+
+    Calculates the cumulative distribution of a value within a window or
+    partition. The cumulative distribution is defined as the number of rows
+    preceding or peers with the current divided by the total number of rows in
+    the frame.
+
+``DenseRank``
+-------------
+
+.. class:: DenseRank(*expressions)
+
+    Equivalent to ``Rank`` but does not have gaps.
+
+``FirstValue``
+--------------
+
+.. class:: FirstValue(expression)
+
+    Returns value evaluated at the row that is the first row of the window
+    frame. If no such value exists, a ``null``-value is returned.
+
+``Lag``
+-------
+
+.. class:: Lag(expression, offset=1, default=None)
+
+    Calculates the value offset to ``offset``, and in case no row exists,
+    default is returned. ``default`` needs to have the same type as the field.
+
+    No validation is done on this expression, i.e., the types of ``field`` and
+    ``default`` are not compared.
+
+``LastValue``
+-------------
+
+.. class:: LastValue(expression)
+
+    Comparable to :class:`FirstValue`, the calculates the last value in a given
+    frame clause. The type of the return value is the same as ``expression``.
+
+``Lead``
+--------
+
+.. class:: Lead(expression, offset=1, default=None)
+
+    Calculate the leading value in a given frame. The fallback ``default`` must
+    be of the same type as ``element``. Both offset and default are
+    evaluated with respect to the current row.
+
+    No validation is done on this expression, i.e., the types of
+    ``expression`` and ``default`` are not compared.
```

Comment:
Please wrap docs at 79 characters, i.e.:
```
    Calculate the leading value in a given frame. The fallback ``default`` must
    be of the same type as ``element``. Both offset and default are evaluated
    with respect to the current row.

    No validation is done on this expression, i.e., the types of ``expression``
    and ``default`` are not compared.
```

Reply:
I formatted the other doc-file, but clearly forgot this. Thanks.